content/en/blog/_posts/2023-10-18-gateway-api-time-to-switch-i2gw/index.md

```diff
@@ -0,0 +1,111 @@
+
+### Expressiveness
+The Gateway API sets Kuberentes standard support of a wide range of features, such as header-based matching, traffic splitting, weight-based routing, request mirroring and more. With Ingress, these features need custom provider-specific annotations.
+
+### Extensibility
+Gateway API allows for custom resources to be linked at various layers of the API. This makes granular customization possible at the appropriate places within the API structure.
+
+## Getting Started
+
+### Gateway API Quick Start
+
+The first thing you will need is a Kubernetes cluster. You can set up a new cluster using [KinD](https://kind.sigs.k8s.io/) with the following command:
+```bash
+kind create cluster --name introducing-i2gw
+```
+When you create a service of type LoadBalancer in a Kubernetes cluster, typically, cloud providers (like AWS, GCP, Azure) automatically provision a cloud load balancer and allocate a public IP address for it. However, since KinD runs locally, there's no cloud provider available to automatically provision an external IP for such services.
+
+This is where MetalLB comes in. MetalLB provides a network load-balancer implementation for Kubernetes clusters that do not have external load-balancers available, like KinD.
+You'll need to [install](https://metallb.universe.tf/installation/) it in order to continue with this guide.
+
+Now you should have a working environment to start experimenting with Gateway-API. There are many great guides you can follow at https://gateway-api.sigs.k8s.io/guides/.
+
+### Migrating from Ingress to Gateway API using Ingress2Gateway
+Migrating from Ingress to the Gateway API may seem intimidating, but luckily Kubernetes SIG-Network provides a tool to simplify the process. Ingress2Gateway assists in the migration by converting your existing Ingress resources into Gateway API resources. To get started with ingress2gateway, you need to first install the tool.
+
+```
+go install github.com/kubernetes-sigs/ingress2gateway@latest
+```
+This will install ingress2gateway to $(go env GOPATH)/bin/ingress2gateway.
+
+Alternatively, you can build it from source using https://github.com/kubernetes-sigs/ingress2gateway#build-from-source.
+
+Once the tool is installed, you can use it to convert the ingress resources in your cluster to GatewayAPI resources.
+
+```
+ingress2gateway print
+```
+
+This above command will:
+
+1. Read your Kube config file to extract the cluster credentials and the current active namespace.
+1. Search for ingresses and provider-specific resources in that namespace.
+1. Convert them to Gateway-API resources (Currently only Gateways and HTTPRoutes).
+For other options refer run the tool with `-h` or refer to https://github.com/kubernetes-sigs/ingress2gateway#options
+
+Review the converted Gateway API resources, validate them, and then apply them to your cluster.
```

Comment:
I think ideally you'd want to pair this with some guidance such as how you'd use this with production workloads. 

Key points that are currently missing:

1. Follow the [Gateway API Getting Started Guide](https://gateway-api.sigs.k8s.io/guides/) to install the CRDs and an implementation of your choice.
2. Ingress and Gateway can run side-by-side, we recommend running both in parallel and only switching over DNS once you've been able to confirm that routing via Gateway API is working well.

Reply:
> Ingress and Gateway can run side-by-side, we recommend running both in parallel and only switching over DNS once you've been able to confirm that routing via Gateway API is working well.

We have spoken about it before I left for vacation. I think we should publish and official guidance in the gw official website, putting it here is not enough. Whats the go to solution we recommend for dns? I assuming it is [external-dns](https://github.com/kubernetes-sigs/external-dns)?

Asking so I can give references from here

Reply:
changed to reflect this. Resolving for now but open again if you need